src/OpenTelemetry.Instrumentation.AWSLambda/Implementation/AWSLambdaResourceDetector.cs

```diff
@@ -14,12 +14,12 @@ internal sealed class AWSLambdaResourceDetector : IResourceDetector
     /// <returns>Detected resource.</returns>
     public Resource Detect()
     {
-        var resourceAttributes = new List<KeyValuePair<string, object>>
+        var resourceAttributes = new List<KeyValuePair<string, object>>(4)
```

Comment:
Resource requires `<string, object>` and not `<string, object?>`, so I went with setting the values to `string.Empty` though they would be `null` in TFMs without NRT I believe.

Alternatively, I could change this to only add the values to the list if they aren't null, but that might trip people up if they didn't use a guard like `TryGetValue()`.

Reply:
Sending empty strings is not a convention that is usually applied in OTel, so I would go with only adding the values if they are not null.

Reply:
Addressed in #1545.